upstream mirror to 10-15

Hatterhat opened the pull request:
## About The Pull Request

Tin from https://github.com/tgstation/tgstation/commit/d4d5a9a9acab52dfa0acbc0f7a0f0c6d9d78f667 to https://github.com/tgstation/tgstation/commit/7007a304d3daeba97f02385a4f18846eecbc0848.

Fixes soup pots being inaccessible.

Include tgstation/tgstation#92119.

Closes NovaSector/NovaSector#6223 and NovaSector/NovaSector#6224.

## How This Contributes To The Nova Sector Roleplay Experience

Continued upstream parity.

## Proof of Testing

<img width="296" height="194" alt="image" src="https://github.com/user-attachments/assets/7a661250-19fd-44e7-aa22-5d4450147301" />

## Changelog

Included.

Sothanforax committed:
Moth Emotes for the masses. (#93282)

tgstation-ci[bot] committed:
Automatic changelog for PR #93282 [ci skip]

Ghommie committed:
Removed duplicate scream emote. (#93104)

We have two different *scream emote types, not counting the *screech
subtype, and it's a bit of a copypasta, so I merged the two together,
and made the *screech its own type, only usable by human mobs ~~with the
primitive traits, aka monkeys and ashwalkers~~ like before.

Less copypasta. You'll probably have to reset your scream hotkey after
this is merged.

~~Also screeching is simply another emote that played the scream sound,
and human scream stock sounds barely fit it anyway, so we're not losing
much by locking it to monkeys and ashwalkers.~~ Screeching is once more
back on the menu

:cl:
code: Merged the human and living versions of the scream emote. Remember
to reset your scream hotkey if it is ever unset.
/:cl:

tgstation-ci[bot] committed:
Automatic changelog for PR #93104 [ci skip]

necromanceranne committed:
Adds the Jaws of Recovery, a specialized form of Jaws of Life for paramedics with heavier limitations. (#93021)

Adds the Jaws of Recovery, a form of Jaws of Life. These spawn in
cabinets in medical, similar to the fireaxe and mech removal tool.

Jaws of Recovery have two heads; prying, like a standard set of jaws of
life, and bonesetting.

Jaws of Recovery cannot be used to open windoors, and cannot be used to
open certain restricted doors. These doors include command staff private
offices, any command specific areas, AI upload areas and security areas
that aren't the brig entrance.

Jaws of Recovery also send out an alarm whenever used to open a door
that is of meaningful significance, like a departmental area. This does
not happen when opening maintenance airlocks, public accessible doors,
external airlocks and the auxiliary base.

The standard Jaws of Life and Syndicate Jaws of Death are entirely
untouched and function as expected.

<img width="240" height="177" alt="image"
src="https://github.com/user-attachments/assets/4e661720-25c7-42b5-963d-707b77d3683f"
/>

In my last PR I removed the broad access available to paramedics.
https://github.com/tgstation/tgstation/pull/92751
I have already explained my reasoning as to why this broad access is a
mistake. This is not anywhere close to that broad access being returned.
This is a slow, deliberate interaction meant to create friction when
someone is moving around using this tool.

Now, obviously, the change was something of a kick in the teeth for
paramedics (which I won't apologize for doing). I hadn't received any
moderate alternatives that sounded like a good idea for months. That is,
until the PR was merged and someone told me about how paramedics over on
Baystation have jaws of life, but they alert people over the radio when
they are used. Seemingly they were as much worried about paramedic
tiders as we are, and they're a high roleplay environment. We clearly
invited a problem on our end that they had sought to resolve because
they too realized that it was a mistake to just give the role this
access.

Now, obviously, [RETA exists to get medical staff into an
area](https://github.com/tgstation/tgstation/pull/92753). That's all
well and good and I like that system. This does not detract from that PR
whatsoever either. This is meant to serve as an extra emergency measure
in the event a paramedic REALLY needs to get into an area while trying
to create some kind of accountability for their actions by telling sec
when they start being used to pry open airlocks. If sec doesn't like how
much the paramedic is invading areas without reason, they at least know
where and when they've been going into locations and might be
incentivized to go investigate.

:cl:
add: Adds the Jaws of Recovery. One can be found in a locked glass
cabinet in medical. These jaws of life are restricted on what they can
force open, and alert security and medical whenever they are utilized to
pry open departmental doors. They can't be used to open high security
doors whatsoever. And they come with a bonesetter attachment instead of
wirecutters. You know, for power-relocating your arm.
/:cl:

tgstation-ci[bot] committed:
Automatic changelog for PR #93021 [ci skip]

Fghj240 committed:
A new type of cybernetic ear, volume-adjusting ears. Also changes the stats of the basic and regular cybernetic ears for consistency (#93362)

tgstation-ci[bot] committed:
Automatic changelog for PR #93362 [ci skip]

MrMelbert committed:
Artifically spawned mobs have a vastly reduced penalty for death moodlet (#93348)

## About The Pull Request

Closes #93285

Mood event from the death of a mob spawned "artificially" is 75% weaker,
lasts 80% the duration, and don't compound

An artificial monkey's death will now result 
- -8 * 0.25 * 0.5 = floor(1) = 1 strength moodlet for the average crew
member
   - ...Lasting 30 seconds (unless refreshed)
- -8 * 0.25 * 1.5 = floor(3) = 3 strength moodlet for animal friends
   - ...Lasting 1.5 minutes (unless refreshed)
- -8 * 0.25 = floor(2) = 2 strength moodlet for compassionate crew
members
   - ...Lasting 1 minute (unless refreshed)

Artifical spawning includes
- Moneky Cube
- Xenobiology Console
- "Life" reaction
- Summoned rats
- Spawner grenades
- Cult ghosts

Lemmie know if I'm missing any obvious spawns 

## Why It's Good For The Game

While funny it was not my intention to have Xenobiology / Genetics /
Virology nuke your mood.

## Changelog

:cl: Melbert
balance: Death of artifical mobs (such as monkey cube monkeys) result in
a 75% weaker, 80% shorter moodlet that does not compound with more
deaths.
/:cl:

necromanceranne committed:
Restores the base warp_effect object (#93368)

## About The Pull Request

Was deleted in https://github.com/tgstation/tgstation/pull/92216

Should not have been.

## Why It's Good For The Game

Deleting used assets bad.

## Changelog
:cl:
fix: The warp sfx now works properly.
/:cl:

MrMelbert committed:
Fix soup pots being unreachable (#93377)

## About The Pull Request

Caused by #93165 

I'm not sure but I think this was just missing a check for
`TRAIT_SKIP_BASIC_REACH_CHECK`, because soup pots are given that trait

## Changelog

:cl: Melbert
fix: You can interact with soup pots on stoves again
/:cl:

lelandkemble committed:
Allows head-equippable items that are not /obj/item/clothing to be attached to /datum/component/hat_stabilizer (#93345)

<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may
not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull
request process. -->

## About The Pull Request

the `/datum/component/hat_stabilizer` component, which is the thing on
the magnate MODsuit(and apparently on other spacesuits(and wigs)) that
allows you to attach an additional piece of headwear on top of the helm,
was `istype()`ed to only allowing `/obj/item/clothing`. This prevented
many non- `/clothing` items from being attached, despite being
equippable to the head, the only one of which I care about being
flowers.

This pr changes the `istype()` check to checking for the head slot flag,
and modifies surrounding logic to account for this. The only items I can
think of that are non-clothing and are equippable to the head are
various flowers and `mob_holder`s for various animals.

Also fixes an unrelated bug where updating the appearance of the holding
item would change the offset of any loose-hat holders over and over
again, causing it to move around over your head if you, for example,
toggled a MOD flashlight on and off repeatedly.

Also fixes an unrelated bug where the hat taken off would never be put
in your hands when it was seemingly intended to, due to always returning
the hat rather than `put_in_active_hand()` (unless the parent isn't
clothing, but there's no non-clothing items with the component attached)

## Why It's Good For The Game

I want to put flowers on my helmet, and I don't see any possible
negative ramifications.

## Changelog

:cl:

qol: Flowers can now be attached to space helmets.
qol: Taking hats off of helmets will now put them in your hand.
fix: MOD flashlights won't move hats on them when toggling.

/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put
your name to the right of the first :cl: if you want to overwrite your
GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the
icon ingame) and delete the unneeded ones. Despite some of the tags,
changelogs should generally represent how a player might be affected by
the changes rather than a summary of the PR's contents. -->

tgstation-ci[bot] committed:
Automatic changelog for PR #93348 [ci skip]

tgstation-ci[bot] committed:
Automatic changelog for PR #93368 [ci skip]

tgstation-ci[bot] committed:
Automatic changelog for PR #93377 [ci skip]

tgstation-ci[bot] committed:
Automatic changelog for PR #93345 [ci skip]

timothymtorres committed:
Fix lighters triggering runtimes when out of fuel (#93387)

tgstation-ci[bot] committed:
Automatic changelog for PR #93387 [ci skip]

timothymtorres committed:
Water helps heal liver slowly (#93367)

Co-authored-by: Lucy <lucy@absolucy.moe>

tgstation-ci[bot] committed:
Automatic changelog for PR #93367 [ci skip]

vinylspiders reviewed (changes requested):
Good work, only thing I would say is for the jaws icons, those are just the TG ones right? instead of copy pasting into the 'aesthetics' .dmi they it would be better if the subtype were de-overridden to point back to TG.

Like how they do it here (except with the inhand ones de-overridden as well) :

https://github.com/NovaSector/NovaSector/blob/6530592544aa57c26312684a0a79d2c38b4546bc/modular_nova/modules/aesthetics/tools/code/tools.dm#L23-L24

The "on" state that was missing from TG which you added here should be PR'd upstream.

Also I am noticing we have some (old?) combat wrench icons in here too. All of those should be deleted from aesthetics and the same thing should be done with `/obj/item/wrench/combat`.

eg

```
/obj/item/wrench/combat
	icon = 'icons/obj/tools.dmi'
```

<img width="780" height="320" alt="image" src="https://github.com/user-attachments/assets/f174acea-877e-48a1-9921-0cc5ad13713d" />

Hatterhat committed:
icon deduplication

paganiy committed:
Changeling's last resort breaks near obstacles (#93109)

## About The Pull Request

- **fixes** a runtime error that occurred when removing antag info
button from the ling's old body (which became null) after transforming
into a worm.
- **fixes** a bug that caused a worm to appear in the debug room when
spamming the "yes" button in the TGUI alert window
- **changes** the blind range of last resort from 2 to 4, because it has
a short animation now

**Added** the ability for the changeling's "Last Resort" to break
through walls and other objects accompanied by blood, gore, and
disgusting sounds😊


https://github.com/user-attachments/assets/a2f263ec-a809-404f-a001-8d3643ba887b
## Why It's Good For The Game
currently, the explosion from "last resort" feels generic, similar to a
grenade or bomb. As a bioweapon, the changeling's ability should reflect
its biological nature — as if the ling is rapidly expanding its cells,
bursting its own body to rupture the surrounding environment and
explode.

regarding the ability to break walls: the current design allows for a
counter-play strategy of quickly building walls around a ling (it takes
like 10 seconds or less) who is in stasis or accumulating reagents for
last resort. While containing a ling in proper cell is valid roleplay
(like for science), this "fortnite-style" walling tactic feels cheap.
The intended design of the ability is to force the destruction of the
ling's body, not to allow them to be permanently walled in with no
option other than ghosting.

Yeah, i know that players will just build a cell that's 2 walls thick or
even bigger, but i think that's fiiine.
## Changelog
:cl:
add: changeling's last resort ability now can rupture surrounding them
walls and other objects
add: last resort now has a short animation
balance: increased last resort's blind range from 2 to 4 tiles
fix: fixed several runtimes related to changeling's last resort ability
sound: added a gore sound effect to last resort explosion
/:cl:

tgstation-ci[bot] committed:
Automatic changelog for PR #93109 [ci skip]

MrMelbert committed:
Fix modsuits and defibs (#93373)

Fixes #93359

Caused by #93165

Inventory screen elements were no longer considered reachable, which
broke mousedrop handing on objects that check "is dragging into
inventory slot"

I don't know the best way to fix this yet but I figured the next best
thing would be to make all of these use the `drag_pickup` element, which
skips this reach-ability check

Thus I refactored it slightly to accommodate for items which should
contextually not be drag-pick-up-abble and bam, works like a charm

:cl: Melbert
fix: Dragging defibs and modsuits off your back works again
/:cl:

---------

Co-Authored-By: ArcaneMusic <41715314+ArcaneMusic@users.noreply.github.com>
Co-Authored-By: Xander3359 <66163761+Xander3359@users.noreply.github.com>

tgstation-ci[bot] committed:
Automatic changelog for PR #93373 [ci skip]

Krysonism committed:
Fixes living food / creature hybrids not grabbing ghosts or storing brains. (#93405)

## About The Pull Request

This PR fixes a long standing issue where food creatures created with
player brains did not work correctly, due to their code checking for the
brain within their contents, where it was absent.

It also stores the heart used in contents for consistency. 

## Why It's Good For The Game

It lets people enjoy playing as these wonderous creatures, and lets
these strange beasts be revived when destroyed with proper veterinary or
culinary intervention.

## Changelog

:cl:
fix: food / animal hybrids like the butterbear and cak now once again
correctly grab ghosts and store brains.
/:cl:

tgstation-ci[bot] committed:
Automatic changelog for PR #93405 [ci skip]

lelandkemble committed:
Gets rid of lie in contractor kit uplink purchase description (#93400)

## About The Pull Request

It says there's three random items, but that isn't true anymore.
Also, some messed-up spacing.

## Why It's Good For The Game

The syndicate's evil and all, but lying might be a bit too much

## Changelog
:cl:

spellcheck: Removed lie about free stuff from contractor kit uplink
description

/:cl:

necromanceranne committed:
Restores a missing sprite on the jaws of recovery + some empty cases (#93399)

## About The Pull Request

Somehow this sprite ended up going missing, probably while resolving
merge conflicts. This restores the sprite for the bonesetting mode.

Also some of the cabinets were mapped in empty. Oops.

## Why It's Good For The Game

I got told this the moment I woke up, and when I sat down to fix it, the
pr was merged. AAAAAH

## Changelog
:cl:
fix: The Jaws of Recovery bonesetting mode now has an actual sprite.
fix: Some Jaws of Recovery cabinets that were strangely empty are now
filled.
/:cl:

tgstation-ci[bot] committed:
Automatic changelog for PR #93400 [ci skip]

tgstation-ci[bot] committed:
Automatic changelog for PR #93399 [ci skip]

necromanceranne committed:
Turns the chaplain's energy swords into...energy swords. (#93394)

## About The Pull Request

The chaplain's energy swords were weird fake energy swords that couldn't
be turned on or off. This is largely because, in the past, we couldn't
easily convert other objects into null rods conveniently. Though we did
get the transforming component, it never was added to the chaplain
weapons. Maybe because there would be a lot of copypasted code.

Now, with the ``nullrod_core`` element, we can easily turn an object
into a null rod. So I did that. It's now an energy sword AND a null rod.
The future is now.

For clarity, the stats on the null rods are identical to before. The
only difference being that it has the ability to switch on and off, and
it now emits light. That's it really. If there are any extra stats I
missed, let me know.

Also adds the carp'sie plushie to the list of wearable objects for
chaplain suit storage.

## Why It's Good For The Game

This has bugged me like, for a really, really long time. And we now have
a lot of the bits and pieces to make more of the null rods actually
closer to the objects they're ripping off.

## Changelog
:cl:
qol: The chaplain energy swords now work like energy swords. Turn them
on and off! Light your way with the blade. Light a cigarette with your
blade like a badass. Feel bad that it isn't a REAL energy sword.
fix: The carp'sie plushie now fits in chaplain suit storage.
/:cl:

tgstation-ci[bot] committed:
Automatic changelog for PR #93394 [ci skip]

archbtw0 committed:
Fixes riot shotgun unintentional melee when point blank shooting (#93320)

## About The Pull Request

Due to 'fire_delay = 8', when pump racking a riot shotgun (with a live
shotgun bullet in it) it will instead melee when you try to point blank
shoot at someone.
This PR fixes that by blocking melee, on the condition that you pump
racked the riot shotgun and it is about to be ready to fire.

You still melee with LMB if you have no ammo pump racked/didn't pump
rack the shotgun, and if you have ammo pump racked, you can use RMB as
intended.

tldr Melee combat functionality for riot shotgun remains unchanged, the
PR only fixes the unintentional melee attack.

## Why It's Good For The Game

The PR fixes unintentional tediousness when wanting to fire at someone
point blank with a riot shotgun - It will no longer melee, which is
good, because otherwise it makes arbitrarily make riot shotgun cooldown
longer due to an unintentional melee attack.

## Changelog

:cl: ArchBTW
fix: Fixes riot shotgun unintentional melee when point blank shooting
/:cl:

tgstation-ci[bot] committed:
Automatic changelog for PR #93320 [ci skip]

SmArtKar committed:
Sleeping Carp and Cain & Abel no longer tell you about armor penetration when you reflect projectiles with them (#93275)

## About The Pull Request

Projectile refactor pulled armor check above the pre-hit comsig, this
fixes that. No need to check armor before you're hit when you
potentially will not be.

## Changelog
:cl:
fix: Sleeping Carp and Cain & Abel no longer tell you about armor
penetration when you reflect projectiles with them
/:cl:

tgstation-ci[bot] committed:
Automatic changelog for PR #93275 [ci skip]

Gaxeer committed:
refactor detective scan categories, fix some categories not being shown at all, remove Syndicate scan category (#93138)

## About The Pull Request

Refactor detective scan categories.
Fix some categories not being shown at all.
Remove Syndicate scan category.

<details>
<summary>TGUI</summary>
<img width="514" height="545" alt="image"
src="https://github.com/user-attachments/assets/754ce576-545e-40e5-9fad-86f19d8d2606"
/>
</details>

<details>
<summary>Paper report</summary>
<img width="460" height="609" alt="image"
src="https://github.com/user-attachments/assets/e37e1965-e90a-4e0c-8109-f4384d6a4a79"
/>
</details>


## Why It's Good For The Game

DETSCAN_CATEGORY_SETTINGS, DETSCAN_CATEGORY_HOLY,
DETSCAN_CATEGORY_ILLEGAL and DETSCAN_CATEGORY_NOTES are now shown in
detective scanner ui and printed report.

Cleaner code

## Changelog

:cl:
fix: energy dagger pen report will be now shown in detective scanner and
also all items that transform will have additional note about it
/:cl:

---------

Co-authored-by: Jeremiah <42397676+jlsnow301@users.noreply.github.com>

tgstation-ci[bot] committed:
Automatic changelog for PR #93138

SmArtKar committed:
[NO GBP] Fixes raptors going at mach speeds (#93418)

## About The Pull Request

I'm pretty sure I tacked on this bit ***after*** testing the PR and
forgot to check if it actually returned the value I needed. Oops.
Closes #93389

## Changelog
:cl:
fix: Fixed raptors going at mach speeds
/:cl:

tgstation-ci[bot] committed:
Automatic changelog for PR #93418 [ci skip]

Ghommie committed:
[NO GBP] Fixes dimensional anomalies duplicating airlocks (#93403)

## About The Pull Request
The code assumed that req_access and req_one_access weren't null. A
mistake that led to runtime errors, preventing the old airlock from
being deleted among other things.

## Why It's Good For The Game
This will fix https://github.com/tgstation/tgstation/issues/93355.

## Changelog

:cl:
fix: Fixed dimensional anomalies duplicating airlocks
/:cl:

tgstation-ci[bot] committed:
Automatic changelog for PR #93403 [ci skip]

vinylspiders committed:
Fixes backwards istype() check (#93420)

## About The Pull Request

Came across this randomly, looks weird, and I think it always just would
evaluate to FALSE which is definitely not the intention.

## Why It's Good For The Game

Fixes a bug

## Changelog

:cl:
fix: integrated toolset and integrated paperwork implants can be emagged
for an integrated combat knife and chameleon stamp, respectively (fixed
a bug preventing that from working)
/:cl:

tgstation-ci[bot] committed:
Automatic changelog for PR #93420 [ci skip]

Cyberboss committed:
Fix Cardinal Sin (#93423)

Of using YYYY/DD/MM what the hell???

Also apparently that table wanted to be auto-formatted. Any objections?

Co-authored-by: Jordan Dominion <Cyberboss@users.noreply.github.com>
# Conflicts:
#	README.md

MrMelbert committed:
Cuffsnapping doesn't spam runtimes in context (#93411)

## About The Pull Request

Asserts that target is a mob when target is definitely not always a mob

<img width="673" height="200" alt="image"
src="https://github.com/user-attachments/assets/f47c2464-0eb1-4bdf-b39f-a35ec53fff2b"
/>

SyncIt21 committed:
Fixes 2 order console bugs (#93417)

## About The Pull Request
- Fixes #93416
- Order console displays correct Private/Department order text amounts

## Changelog
:cl:
fix: order console correctly differentiates between private & department
orders
fix: order console checkout cart displays correct Private/Department
order text amounts
/:cl:

---------

Co-authored-by: ArcaneMusic <41715314+ArcaneMusic@users.noreply.github.com>

tgstation-ci[bot] committed:
Automatic changelog for PR #93417 [ci skip]

SyncIt21 committed:
Fixes circuit multitool revealing special items (#93419)

## About The Pull Request
- Fixes #93409

Code copied from strip menu so the same principles apply

## Changelog
:cl:
fix: circuit multitool won't reveal special items on mobs
/:cl:

tgstation-ci[bot] committed:
Automatic changelog for PR #93419 [ci skip]

TealSeer committed:
Fix theoretical hard delete with redirect_attack_hand_from_turf component (#93412)

## About The Pull Request

- Nulls the ref to the interact callback in
`/datum/component/redirect_attack_hand_from_turf/Destroy()`
- Move parent signal registration from `Initialize()` to
`RegisterWithParent()`
- Add parent signal un-registration in `UnregisterFromParent()`

## Why It's Good For The Game

Moving signal registration was just a "while I'm here" type thing
because it should go in those procs, the ref null is because it's
causing flaky hard deletes [but only on a
downstream](https://github.com/effigy-se/effigy/actions/runs/18415754211/attempts/1)
for some reason, I have no idea why I haven't seen it here but it should
be nulled regardless so

## Changelog

N/A

MrMelbert committed:
Unified CMO blues slightly, updates garment bags slightly, updates CMO locker contents slightly (#93321)

Before - after

<img width="533" height="324" alt="image"
src="https://github.com/user-attachments/assets/c8a843a6-875d-4796-8c62-d24dd509147b"
/>

Logical follow up to #93313 and #91902

CMO now has a backpack which matches their teal palette rather than
re-using the medical doctor's blue backpack

CMO now starts with white sneakers instead of blue sneakers

Garment bags can hold backpacks, duffels, etc.

However they CANNOT store backpacks which have ANY contents, so no risk
of storage nesting memes here.

Removed the defib mount from the CMO's closet

Minor bugfix for being unable to swap storage positions of items which
were instantiated in a storage

- Much like the Paramedic, the CMO has had many cooks which have not
always been on the same page, so this makes their sprite a bit more
coherent.

- Heads with unique packs just tossed them in their closet which is
pretty spammy given we have four variants now. Putting them in the
garment back makes sense.

- Defib mount in CMO's locker hasn't been necessary for a while now that
all treatment centers spawn with defib mounts.

- Bugfix.

:cl: Melbert
image: CMO now has unique backpack/satchel/etc. sprites which matches
their teal palette.
image: CMO spawns with white sneakers instead of blue sneakers.
qol: Garment bags can hold empty backpacks/satchels/etc.
del: Defib mount removed from CMO's locker.
fix: Items stocked in a storage item by default can be
mouse-drag-swapped like any other item.
/:cl:

your bag's deduplicated now. that's my attack
your bag's blue now. that's my attack

tgstation-ci[bot] committed:
Automatic changelog for PR #93321 [ci skip]

improvedname committed:
Adds red alert helper to sec eva on deltastation (#93298)

This pr makes it so sec eva on deltastation is all access when its red
alert
One Alternative solution to #93293 (Part 2 edition)
## About The Pull Request
## Why It's Good For The Game

So officers can get the space suits without breaking in and asking
mister warden whenever its red alert for auctall emergency time based
stuff

## Changelog
:cl: Ezel
map: Deltastation Sec eva becomes all access when its red alert.
/:cl:

improvedname committed:
Sec eva on wawa/nebula/catwalk/icebox now are all access on red alert (#93299)

## About The Pull Request
Makes sec eva be all access on red alert
One Alternative solution to
https://github.com/tgstation/tgstation/pull/93293 (Part 3 edition)

## Why It's Good For The Game

So officers can get the space suits without breaking in and asking
mister warden whenever its red alert for auctall emergency time based
stuff

## Changelog
:cl: Ezel
map: Sec eva now becomes all access on red alert on
nebula/catwalk/icebox/wawa
/:cl:

tgstation-ci[bot] committed:
Automatic changelog for PR #93298 [ci skip]

tgstation-ci[bot] committed:
Automatic changelog for PR #93299 [ci skip]

Hatterhat committed:
syndicate modsuit visor pseudo-toggle (tied to combat mode) (#93391)

The visors on MODsuits using the Syndicate welding visor (now renamed
the "flash-protected optical suite") now toggle their armor booster
overlay depending on whether the user has combat mode on or off.

Style. Listen, man, I just thought it'd be a shame if the cool visor
sprites never got used again.

https://github.com/user-attachments/assets/d5103168-7a1e-4405-adb3-53420c8601d7

:cl:
add: The visors on MODsuits that used to have armor boosters now visibly
toggle based on whether the wearer is in combat mode or not.
/:cl:

---------

Co-authored-by: Hatterhat <Hatterhat@users.noreply.github.com>
Co-authored-by: MrMelbert <51863163+MrMelbert@users.noreply.github.com>

tgstation-ci[bot] committed:
Automatic changelog for PR #93391 [ci skip]

Xander3359 committed:
Converts more attackby's to interactions (#93106)

## About The Pull Request
Converts the following:
- Medical Kiosk
- Implant case
- Flamethrower
- Chemical implant case
- Pappercutter

Also I've looked at some alt click procs and adjusted some of their
returns

tgstation-ci[bot] committed:
Automatic changelog for PR #93106 [ci skip]

FalloutFalcon committed:
ghost basic mobs use natural hair colors (#93437)

<img width="1142" height="956" alt="image"
src="https://github.com/user-attachments/assets/6395e71d-cecd-42f1-a0b7-ebf322a3cb65"
/>

I have no whimsy and am repulsed by the funny colors (standardization
for how the rest of human hair color selections is done.)
:cl:
fix: ghost basic mobs use natural hair colors
/:cl:

tgstation-ci[bot] committed:
Automatic changelog for PR #93437 [ci skip]

Fghj240 committed:
Non-admin spawned atlas gravitronic spinal implants now use their intentional(?) sprite (#93428)

## About The Pull Request

so apparently there are two sprites for herculean gravitronics, one with
the core in and one without, but the code for adding a core to the
gravitronic doesn't actually change the sprite so it's unused unless an
admin spawns in an atlas with a core already in it which in that case
it'll have the sprite.

This single line of code makes the sprite change when a new core is put
in.

yes I tested it no it's not a web edit

## Why It's Good For The Game

bugfix

## Changelog

:cl:
fix: Atlas gravitronic spinal implants now use their intended sprite
/:cl:

Co-authored-by: Fghj240 <fakeemail@notrealemail.com>

tgstation-ci[bot] committed:
Automatic changelog for PR #93428 [ci skip]

necromanceranne committed:
Increases the experience earned by Hunter Boxing (#93435)

## About The Pull Request

Hunter boxing now uses the full quantity of damage inflicted by your
punches, rather than half. We also multiply our experience based on the
max health of the creature we're punching.

You do not get the exercised status effect for punching mobs with Hunter
Boxing.

You do not earn experience for punching dead things.

Hunter Boxing can now brutalized Crustaceans and Reptiles. This does not
include lizardpeople, as they're humanoid. No, you can't Hunter Boxing
the ashwalkers. Or your lizard QM boss.

## Why It's Good For The Game

It takes a really, really, really long time to level up using Hunter
Boxing. Normal athletics progression is largely kept in line by the fact
that you need to go sleep every time you do a workout session.

Hunter boxing, however, does not allow you to do that because you're
currently trying to punch xenofauna and demons to death while in the
middle of a hell planet and trying not to die.

The fact that fauna fight back when punched means leveling up with
hunter boxing has its own internal complications for preventing
athletics progression from being a walk in the part. Obviously, the
threat of death is its own punishment mechanic.

It's maybe a bit much of an ask miners to be weightlifting when they
should be mining after all.

## Changelog
:cl:
balance: Hunter boxing now grants a reasonable amount of experience in
Athletics for punching the absolute shit out of the local wildlife.
balance: Includes crustaceans and reptiles to the list of mob types that
Hunter Boxing is allowed to mangle with their bare hands. You still
can't use Hunter Boxing on lizardpeople. Or crabpeople when they are
added in 2030.
/:cl:

tgstation-ci[bot] committed:
Automatic changelog for PR #93435 [ci skip]

Yobrocharlie committed:
Buffs the cleric mace (#93385)

tgstation-ci[bot] committed:
Automatic changelog for PR #93385 [ci skip]

archbtw0 committed:
Adds Stunsword (#93270)

Co-Authored-By: necromanceranne <40847847+necromanceranne@users.noreply.github.com>

tgstation-ci[bot] committed:
Automatic changelog for PR #93270 [ci skip]

tontyGH committed:
Moves computer frame's anchored check to frame finalization (#93446)

## About The Pull Request

- Moves the anchored check to when construction is being finalized
- Removes unnecessary "missing components" balloon alert when failing
finalization

## Why It's Good For The Game

Fixes a bug where you could have unanchored functional PCs which can be
pretty powerful depending on the PC. It's fun so I hate it and it needs
to die instantly.

## Changelog

:cl:
fix: moves the computer frame's anchor requirement from circuit step to
finalization step, fixing an issue where you could construct unanchored
and FUNCTIONAL computers
qol: as a side effect, you only need to wrench the PC down when
construction is being finished
/:cl:

tgstation-ci[bot] committed:
Automatic changelog for PR #93446 [ci skip]

SmArtKar committed:
Items no longer randomly fall out of bags when changing their size (#93448)

## About The Pull Request
Not sure what this code was meant to achieve, but just checking total
weight is enough. Also added a safeguard to not drop the item if the
storage container was already overloaded just in case we're in a premade
box (this should never happen in-game otherwise)

Fixes #85658

## Changelog
:cl:
fix: Items no longer randomly fall out of bags when changing their size
/:cl:

MrMelbert committed:
Beaker panel TGUI / At least makes it functional (#93398)

## About The Pull Request

Makes Beaker Panel a TGUI so it actually works

<img width="787" height="295" alt="image"
src="https://github.com/user-attachments/assets/343056d2-f059-473d-a1eb-5604d2c3a38e"
/>

There's a lot more I want to do to this UI, namely UX and more features
(like changing grenade types) but I figured I should PR it now that it's
bare minimum functional so I don't get eaten by scope creep

Fixes #89954

## Changelog

:cl: Melbert
admin: Spawn-reagent-container works again, and it's a TGUI
/:cl:

tgstation-ci[bot] committed:
Automatic changelog for PR #93448 [ci skip]

thegrb93 committed:
Fix selfsustaining and clusterbuster bugs with extracts (#93415)

## About The Pull Request
Fixes slime extract bugs with selfstaining slime and clusterbuster.
Fixes: #93315

## Why It's Good For The Game
Fixes slime extract bugs
## Changelog
:cl:
fix: Self sustaining slime and clusterbuster now correctly fill required
reagents
/:cl:

---------

Co-authored-by: Ghom <42542238+Ghommie@users.noreply.github.com>
Co-authored-by: LemonInTheDark <58055496+LemonInTheDark@users.noreply.github.com>

tgstation-ci[bot] committed:
Automatic changelog for PR #93415 [ci skip]

mcbalaam committed:
Fixes and QOL for the "Request Internet sound" verb (#93244)

## About The Pull Request

Fixes inverted `tgui_input` arguments. Prettifies both admin and user
outputs.


https://github.com/user-attachments/assets/e91bc950-0117-42b4-add0-45de2295744d

<img width="687" height="202" alt="image"
src="https://github.com/user-attachments/assets/23b6fa14-016d-4357-ae9b-7121226be4a8"
/>

## Why It's Good For The Game

Cleaner code, looks better ingame

## Changelog

:cl:
qol: Better output for sound requesting
fix: "Request Internet sound" input window's title and description are
not inverted anymore
/:cl:

---------

Co-authored-by: Ghom <42542238+Ghommie@users.noreply.github.com>

tgstation-ci[bot] committed:
Automatic changelog for PR #93244 [ci skip]

vinylspiders committed:
Adds the 'mob_possessor' argument to all the instances of special() (#93410)

## About The Pull Request

a spooky special possession pr for october appears~~

(This is just some code cleanup)
Before, some procs had the `mob_possessor` arg, and some didn't. Which
made it a nightmare to add more positional args to it down the line.
This PR just ensures that all positional args are present in all the
instances of the proc.

## Why It's Good For The Game

Makes this proc chain slightly less of a mess to deal with.

## Changelog

N/A

Fghj240 committed:
The health analyzer can now tell the difference between curable and incurable traumas (#93433)

## About The Pull Request

TRAUMA_RESILIENCE_MAGIC traumas are now called "Persistent" traumas, and
the health analyzer tells you they can be treated with magic.
TRAUMA_RESILIENCE_ABSOLUTE traumas are unaffected.

## Why It's Good For The Game

Makes it easier to understand why some permanent traumas can be cured
while others can't. Also it's kind of weird that a curable trauma is
called "permanent", but also these are two different resiliences and
they shouldn't share a name anyways.

## Changelog

:cl:
qol: Health analyzers now tell the difference between curable and
incurable permanent traumas
/:cl:

Co-authored-by: Fghj240 <fakeemail@notrealemail.com>

tgstation-ci[bot] committed:
Automatic changelog for PR #93433 [ci skip]

TealSeer committed:
Renames flatten_list proc to assoc_to_values (#93453)

`/proc/flatten_list()` -> `/proc/assoc_to_values()`, also changes the
code doc to mirror the language of `assoc_to_keys()`'s code doc

Having a proc called `assoc_to_keys` that takes an associative list and
returns a list of the keys, and also having a proc that takes an
associative list and returns a list of the values, and the latter not
being called `assoc_to_values` is very funny

:cl:
code: renamed flatten_list proc to assoc_to_values
/:cl:

---------

Co-Authored-By: Ghom <42542238+Ghommie@users.noreply.github.com>

tgstation-ci[bot] committed:
Automatic changelog for PR #93453 [ci skip]

Yobrocharlie committed:
Buffs holodeck claymore to 30 force again (#93379)

## About The Pull Request

Reverts the changes in https://github.com/tgstation/tgstation/pull/83334
weak claymore to its former glory so its no longer so bad.

## Why It's Good For The Game

This will let people use melee more often against threats and this
person nerfed the claymore thats used in the holodeck when said claymore
is used elsewhere- and in my opinion the holodeck claymore was not
deserving of this nerf anyhow as it allows crew or antagonists to have a
fairly good melee in a pinch with some extremely hefty downsides such as
the fact that it will disappear from your hand mid fight is non zero.

The other places this claymore can be found is from A: Templar
skeletons, so literally only from the end of snowdin and B: Fishing? if
this becomes a problem ill make a subtype of the claymore that deals 24
brute and 10 pen like how it used to and just give that to the fishing
corpse instead

and also fallowing the format that mothblocks gave me

for a revert to merge, a rule of thumb is it must prove **one** of the
following things:

1| that the problem it was solving was redundantly solved by something
else: The claymores obtainable from the holodeck can be removed from the
holodeck by a law 2 or by one bomb, the holodeck is a massive area and
its very hard to fortify the holodeck against attack and if somebody
does try to make an r-wall around the entire holodeck your going to have
a long notice for that, alternatively you can use x-ray eyes+telepathy
to just turn it off remotely even if its been fortified.

2| that it did not solve the problem, and has either made it worse or
provides an extra loss: It did solve the problem by making this weapon
with already huge downsides completely unviable for any purpose other
than being a shield.

3| that the problem it was solving is illegitimate (the absolute hardest
one): This was an ided PR, its admitted to be such
https://github.com/tgstation/tgstation/pull/83334#issuecomment-2120321699
and they say this which corroborates with the comment that the comment
im linking to is replying to. "Hey AI, I need my free 4-hit weapon. I'm
totally not a changeling or anything." And the issue that they were
"solving" also in-advertently nerfed the one other location where you
could actually get this claymore from. In my opinion these claymores are
not a problem for the reasons as stated above and the reason why it was
ever even merged has a good chance of just being oranges going "haha
this funny"

## Changelog
:cl:
balance: restored the weak claymore to its former glory
/:cl:

tgstation-ci[bot] committed:
Automatic changelog for PR #93379 [ci skip]

FalloutFalcon committed:
skips abstract station traits from admin menu (#93439)

## About The Pull Request
before
<img width="514" height="325" alt="image"
src="https://github.com/user-attachments/assets/e33b48b5-b1ad-4589-be35-fa192081ef78"
/>
fixed
<img width="527" height="515" alt="image"
src="https://github.com/user-attachments/assets/eed5f454-ee52-4b40-98a1-1279a346b894"
/>
## Why It's Good For The Game
By design most of these wont work, i think they cant even be run due to
the subsystem still doing sanity before starting
## Changelog
:cl:
admin: abstract types dont show up in admin selection menu for station
traits
/:cl:

---------

Co-authored-by: Ghom <42542238+Ghommie@users.noreply.github.com>

tgstation-ci[bot] committed:
Automatic changelog for PR #93439 [ci skip]

Ghommie committed:
Flash and flasher refactor/rework (simple mobs no longer immune edition) (#93076)

Co-authored-by: necromanceranne <40847847+necromanceranne@users.noreply.github.com>

tgstation-ci[bot] committed:
Automatic changelog for PR #93076 [ci skip]

SmArtKar committed:
Fixes ore bags being able to reject any ores despite having ample space, stack merging sanity/QOL (#93451)

## About The Pull Request

Ore bags try to insert the stack without actually checking if it can
merge with anything, meaning that ores can take up limited slots which
may end up with 7 single-ore stacks preventing any more ore from
entering the bag, despite there being ample space for 49 more ore of
each type. Now bags will actually try to merge picked up ore into
themselves first, ensuring that this doesn't occur, and that the bag
will always be topped up (This needed additional logic to ensure that
stacks don't merge beyond the size that the storage they're in can
handle)

Additionally, I've changed how stacks merge when dragged (instead of
merging themselves into the target, target is merged into themselves)
and prevent dragging from changing focus to stack that is being merged
into if the merging stack isn't empty, fixing the "pull churn" issue
that locks out player inputs if you try to pry and drag floor tiles due
to constantly swapping the pull target after every tile pried, plus I've
removed that weird ore box `insertion` animation (which I recon is an
oversight) and ore bags sucking up the ores you drop from your hand
because you're probably doing so intentionally

## Changelog
:cl:
fix: Fixed ore bags being able to reject any ores despite having ample
space
fix: Pulling a stack behind you and getting it topped up by lying items
will no longer spam pulls and lock out your inputs
fix: Picking up ores into an ore box via an ore bag will no longer play
a jank animation
qol: Ore bags and MODule ore bags no longer pick up ores you've dropped
from your hand until you walk over them again
/:cl:

tgstation-ci[bot] committed:
Automatic changelog for PR #93451 [ci skip]

grungussuss committed:
item offers now persist when pulling and moving a tile away. (#93252)

tgstation-ci[bot] committed:
Automatic changelog for PR #93252 [ci skip]

EnterTheJake committed:
Heretic Antagonist Full Overhaul. (#92119)

Heretic has received a complete overhaul. This PR touches nearly every
aspect of the antagonist. For readability's sake, not every change is
going to be listed in this pull request.

For the full list of changes please refer to the design doc:
https://hackmd.io/@BiST8PJVRjiwVPY86U3bLQ/B11HyChz1g.

Code by Me, @Xander3359 and @Arturlang

TGUI by @Arturlang

Sprites by OrcaCora and GregorDM

Writing bits by @necromanceranne

- Cross-pathing has been removed. Main knowledge spells are now
exclusive to their path (for the most part).

- For every main knowledge unlocked (save for the robes and the blade
upgrade), Heretics can choose one option from a draft of 3 random side
knowledges (this is a free point).

- Heretics can now purchase side knowledges from a new tab, the
"Knowledge Shop". Side-knowledges have been divided by tier (Stealth,
Defense, Summons, Combat and Main). Tiers are unlocked as you progress
toward your main path.

- Heretics now gain the grasp and mark upgrade immediately, but their
main knowledge choices cost twice as much (except for the first spell,
the robes and the blade upgrade).

- Path specific robes have been introduced! They come with their own set
of quirks.

- Each Path has received a passive ability. This passive is upgraded
when you first create your robes, and again when you complete the Ritual
of Knowledge.

- Paths have been rebalanced as a result of the removal of cross-path
progression. Cosmic and Moon paths have received soft reworks.

- Upon unlocking the path 2nd level or reaching a total of 8 points
worth of knowledge, Heretics will lose the ability to blade break (and
the limit on blades all together).

- Ascension now automatically calls the shuttle with no possibility of a
recall.

- Late join Heretic has been removed.

<img width="750" height="635" alt="moon path ui"
src="https://github.com/user-attachments/assets/184ef783-5c9c-48a1-a2f7-4807ca93e990"
/>

<img width="787" height="669" alt="Knowledge shop"
src="https://github.com/user-attachments/assets/3dc89b84-8c70-4d47-b612-54396e3ea6e7"
/>

- Heretics will now gain X-ray vision for a few seconds when nearby an
eldritch essence (this effect has a cooldown).

- Ritual of knowledge now requires 1 uncommon item instead of 2. You may
now use a stunprod instead of a baton to complete the ritual. Beartraps
have been removed from the list of possible reagents.

- The maximum number of possible sacrifices required to ascend has been
reduced from 6 to 5 while the minimum has been upped to 4.

- Codex Cicatrix no longer requires a special pen to be made.

- Heretics now start with a passive ability. You can find what it does
on the path info tab after a path has been selected, and what they gain
when upgraded.

- Crafting your first set of Eldritch robes will bump your passive to
level 2.

- Unlocking the 2nd level will subsequently unlock your "Ritual Of
Knowledge"

- Completing the ritual of knowledge or ascending will net you the final
level.

- Armorer's Ritual is no longer a side knowledge. Each path will have
their own unique version of the ritual. This is placed after the 2nd
spell in the tree.

- Robes can no longer be destroyed by fire and acid, grant t4 flash
protection (Moth Heretics stay winning) and protection against basic
syringes, to bring them on par with other antagonist's armor sets.

- The recipe to craft the robes is now a set of armor/vest, a mask (any
mask will do now, not just gas masks), plus the unique reagent required
for the blades (Plasma for Cosmic, Trash For Rust, match for Ash and so
on)

- Wearing the robes as a non-heretic may yield some unfortunate
side-effects.

Moon path  rework.

Moon Heretics gain immunity to brain traumas and slowly regenerate brain
health. Equipping the moon amulette channels its effects through the
moon blade; making it unblockable and cause sanity damage instead of
brute. Ring leader's Rise now summons an army of harmless clones that
explode when attacked; the explosion briefly stuns non-heretics and
cause sanity and brain damage to them. Moon blade can also now be used
when pacified and Moon spells are no longer blocked by regular anti
magic, only mind magic protection.

**Cosmic Path Rework**

Cosmic path has received the biggest batch of changes alongside Moon.
The path has been dead last in ascension and pickrate (less than 5%) for
almost 2 years. It did gain some popularity over the last few months,
reaching the highest ascension rate in the game (12%) while mantaining a
relatively low pickrate.

Cosmic sits in a weird spot, where pretty much every knowledge
surrounding the path is either mediocre or, in the case of the
ascension, dysfunctional. Yet it has maintained a smidge of relevancy
due to how quickly Cosmic heretics can capture and sacrifice targets
thanks to Star Touch.

As a result, the best course of action would be to rebalance the
entirety of the kit; granting the heretic more tools to manipulate space
and dictate the flow of a fight, while lessening their ability to end a
confrontation by instantly sleeping their opponents.

lastly The Star Gazer is now ghost controlled ; And they shoot lazers!

<img width="636" height="451" alt="gazer gag 3"
src="https://github.com/user-attachments/assets/601d6881-c042-4e42-8ce6-ac90cd27848b"
/>

Again, if you want my full reasoning, please check my doc
https://hackmd.io/@BiST8PJVRjiwVPY86U3bLQ/B11HyChz1g.

To keep it short and concise; Heretic is too complex and unintuitive for
its own good. Too impenetrable for new players and too abusable for
experienced players. This can be chalked up to a lot of poor design
decisions. But ultimately, what I believe being the biggest contributor
to the current status of Heretic is the ability to move into different
paths, also known as "Cross-Pathing".

Cross-pathing, while cool in theory, overcomplicates the antagonist and
overloads them with power. Players dealing with the heretic are
incapable of working out what a given heretic can do. This also leads to
late game heretics having 3 rows Worth of action buttons and virtually
no weakness.

Over the last year, I've often received the understandable but also kind
of unfair accusations of making Heretic too powerful without a clear aim
or purpose.

My goal with the paths I've reworked over the last year (Rust,Void and
Blade) wasn't necessarily to just make them stronger (although that was
also part of the goal, as they were paths that were underperforming),
but for them to have more interactions with the sandbox and to better
live up to the fantasy presented to the player.

If an harbringer of frost gets countered by a cup of coffee, we probably
messed something up.

Unfortunately, the current incarnation of Heretic doesn't really allow
for surgical balance changes to specific paths. Every time a knowledge
gets buffed, we make every path that can easily tap onto that knowledge
stronger by default. It doesn't take a genius to understand why this
system is ultimately unsustainable.

I feel that after a heretic has reached the near peak of their power,
they no longer need the ability to instantly escape any encounter. Check
my doc for my full reasoning.

By removing cross-pathing, we remove a huge maintainability burden from
the antagonist. Paths can now be designed around clearer strengths and
weaknesses. They become easier to balance and less of an headache to
understand for everyone.

It also means we can give paths some needed quality of life quirks
without having to worry how such a change might have a knock-on effect
for other paths.

Ash heretics can finally let loose without dying by their own flames.
Cosmic Heretic can go to space without having to carry a modsuit. Moon
Heretic can use their abilities without fear of one random trauma
ruining their day, and so on.

people again?

As of right now the heretic tree has quite a hefty amount of trinkets
that pretty much never see use.

Partly because the tree itself is a nightmare to navigate. And partly
because why would anyone set up an elaborate plan or scheme when they
can unleash 2 rows of spell in the span of bunch of seconds.

Heretics mostly gravitate towards powers that push them towards greater,
more potent combat strength. If it doesn't contribute to killing people
quicker, it isn't worth doing for most. And given the opportunity cost
associated for taking those powers, they will remain that way so long as
there are better choices to be poached.

The new draft system encourages Heretics to play more with the tools at
their disposal. If you want to go for a specific combo from the side
path options, you may now do so by tapping into the knowledge shop.

Yes, the shop does include a few knowledges from the other paths. But
these are limited to 1 per path, are very expensive and can only be
unlocked very late into the shift.

The iconic heretic robe is actually sequestered to a side path that is
most easily access by only two paths at a time. Since heretic paths are
being made to be much more specialized, the most obvious way in which
this can be showcased is through an easily
identifiable outfit.

By using the robes, we can both telegraph WHAT heretic you are looking
at, and just how much power they've accumulated and when it is
reasonable to take the kid gloves off and treat them as a genuine
threat. If a heretic is in their
robes, that heretic is now a significantly more prominent danger to the
station.

It also serves as a useful means for gating some of the more powerful
effects of a heretic's path behind the robes, AND enable options for
disarming them of that power should they be captured without making it
something endemic to their mob.

A major problem with heretics is a lack of certainty as to how powerful
they have become. A heretics robes is one of the milestones to help
players dealing with heretics identify that.

This is a massive overhaul to a pretty complex and bloated antagonist.
I've done my best to show the changes to several maintainers and other
members of the community for their feedback. But at some point we'll
have to see how this behave in the environment to get a feel if
something is over or undertuned. (that's my way of saying, yes this is
likely gonna require a testmerge or two).

What I will say is that I'm not trying to change the core identity of
Heretic.

Heretics should have the upperhand in single encounters early on, be
able to joust a small group of players after they unlock their final
spell, and end the round when they ascend. They're a progression
antagonist. They should retain their payoff as well as pose a danger as
they grow stronger.

But if more players feel like they are more reliably able to play the
antagonist in more varied and interesting ways, rather than the
antagonist largely existing as a measuring stick for 'robustness' due to
its elitist design philosophy, then the rework has been a success. There
should be something for
everyone in the antagonist, as is true for all of our antagonist roles.

tgstation-ci[bot] committed:
Automatic changelog for PR #92119
